src/engine.cc

```diff
@@ -77,7 +87,13 @@ void EngineController::PopulateOptions(OptionsParser* options) {
   defaults->Set<int>(Search::kAllowedNodeCollisionsStr, 32);  // Node collisions
 }
 
-SearchLimits EngineController::PopulateSearchLimits(int /*ply*/, bool is_black,
+double move_weight(int ply, float peak, float left_width, float right_width) {
+  const float width = ply > peak ? right_width : left_width;
+  constexpr float width_scaler = 1.518651485; // 2 / log(2 + sqrt(3))
+  return std::pow(std::cosh((ply - peak) / width / width_scaler), -2);
```

Comment:
`-2.0f` to ensure all arguments are float and there are no double <-> float conversion.